torchvision/prototype/features/_bounding_box.py

```diff
@@ -78,13 +78,16 @@ def vertical_flip(self) -> BoundingBox:
 
     def resize(  # type: ignore[override]
         self,
-        size: List[int],
+        size: Union[List[int], int],
         interpolation: InterpolationMode = InterpolationMode.BILINEAR,
         max_size: Optional[int] = None,
         antialias: bool = False,
     ) -> BoundingBox:
         output = self._F.resize_bounding_box(self, size, image_size=self.image_size, max_size=max_size)
-        image_size = (size[0], size[0]) if len(size) == 1 else (size[0], size[1])
+        if isinstance(size, int):
+            image_size = [size, size]
+        else:
+            image_size = (size[0], size[0]) if len(size) == 1 else (size[0], size[1])
```

Comment:
I dislike the nested `if` here. Maybe

```suggestion
        if isinstance(size, int):
            image_size = [size, size]
        elif len(size) == 1:
            image_size = [size[0], size[0]]
```

?

This is almost equivalent with what we had before with the only difference that the previous solution silently ignored `size[2:]`. Meaning, it would still work if we pass sizes of arbitrary size. My suggestion will now raise an exception in the kernel.

Reply:
I tried it but the bbox expected a tuple and this needs to be handled explicitly to make mypy happy. I'll revert the previous approach which just handles the integer and maintains the previous treatment on the else but I'm happy to review on the nits where and how this should be handled. We need to discuss where exactly the exception should be thrown in this case. Possibly on the `new_like` method itself?